src/main/java/org/launchcode/tutorconnector/models/AbstractEntity.java

```diff
@@ -0,0 +1,90 @@
+package org.launchcode.tutorconnector.models;
+
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.MappedSuperclass;
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+import java.util.Objects;
+
+@MappedSuperclass
+public class AbstractEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private int id;
+
+    @NotNull
+    @NotBlank
+    private String firstName;
+
+    @NotNull
+    @NotBlank
+    private String lastName;
+
+    @Email String email;
+
+    @NotBlank
+    @NotNull
+    String pwHash;
+
+    private String timeZone;
```

Comment:
Timezone and Subjects are enums

Reply:
I changed timezone to an enum but subjects we will have to discuss further.